task06.rb

```diff
@@ -0,0 +1,25 @@
+# Convert a string containing phonewords to real phone number, e.g. '1-800-hello' should be converted to '1-800-43556'
+# UPD: We allow dashes, brackets (no need to check the position and if it is a opening-closing pair of brackets), whitespace and "+", but only for the "+" the rule is that it should be the first symbol
+
+def task06(str)
+  if str.class != String
+    raise ArgumentError, "Not a valid format. Need a string."
+  end
+  if str.nil? || str.delete(" ").empty?
+    raise ArgumentError, "String is empty."
+  end 
+
+  str.split("-").each do |s|
+    if s.scan(/\A[a-zA-Z\(\)\+]+\z|\A\s*[+]*\s*[0-9\(\)\+]+\z/).empty?
```

Comment:
@MaraEfimova This regex is a bit incorrect, because ` asd `, ` 123 ` throw an error, while `  123` and ` + 123` do not... And in general I would first sanitize the string (remove whitespace, dashes, etc.) and then do the regex check...